src/api/che-login-manager.ts

```diff
@@ -0,0 +1,531 @@
+/*********************************************************************
+ * Copyright (c) 2020 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+import axios, { AxiosInstance } from 'axios'
+import * as fs from 'fs'
+import * as https from 'https'
+import * as jwt from 'jsonwebtoken'
+import * as path from 'path'
+import * as querystring from 'querystring'
+
+import { ACCESS_TOKEN_KEY } from '../common-flags'
+
+import { CheApiClient } from './che-api-client'
+
+// Represents login information to use for requests
+// accessToken is undefined for single user mode
+export interface LoginData {
+  cheApiEndpoint: string
+  accessToken: string | undefined
+}
+
+// Credentials file format
+export interface CheServerLoginConfig {
+  lastLoginUrl?: string
+  lastUserName?: string
+  logins?: Logins
+}
+
+// API URL -> logins into server
+export type Logins = { [key: string]: ServerLogins }
+// username -> login data
```

Comment:
Maybe it will be better comment `//  username(key) -> login data(value)`?

Reply:
I think it is clear from context. If you don't mind I'll live it as is.